kong/information.go

```diff
@@ -0,0 +1,53 @@
+package kong
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+)
+
+type Info struct {
+	Version       string `json:"version,omitempty" yaml:"version,omitempty"`
+	Configuration struct {
+		Database string `json:"database,omitempty" yaml:"database,omitempty"`
+		Portal   bool   `json:"portal,omitempty" yaml:"portal,omitempty"`
+		RBAC     string `json:"rbac,omitempty" yaml:"rbac,omitempty"`
+	} `json:"configuration,omitempty" yaml:"configuration,omitempty"`
+}
+
+// IsEnterprise check if Kong is enterprise edition
+func (s *Info) IsEnterprise() bool {
+	return strings.Contains(s.Version, "enterprise")
+}
+
+// IsInMemory check if Kong is in memory
+func (s *Info) IsInMemory() bool {
+	return "off" == s.Configuration.Database
+}
+
+// IsRBACEnabled check if RBAC are enabled
+func (s *Info) IsRBACEnabled() bool {
+	return "on" == s.Configuration.RBAC
+}
+
+// IsPortalEnabled check if the portal is enabled
+func (s *Info) IsPortalEnabled() bool {
+	return s.Configuration.Portal
+}
+
+func (c *Client) Info(ctx context.Context) (*Info, error) {
```

Comment:
> And as it would be a single function it doesn't need an entire service of it's own. So it would directly be at the client's level.

There are 3 problems I can think of with this approach:
1. Implementing directly at client's level prevents mocking the info service, and requires the users to operate on a very concrete type with lots of responsibility (remember that `type Client` has all the services as its fields). Better to implement a narrow type that allows querying about info.
2. Once the Admin API has an OpenAPI spec (which can happen soon) we'll be able to replace Kong Admin API struct definitions with autogenerated ones. Making the info service consistent will simplify that process.
3. The root endpoint is a resource like any other, and we should be consisitent with the other getters/setters for other resources.

Therefore please implement an `InfoService` with only the `Get` method on it. I agree that it's more code in this instance, but, in my opinion, the benefits outweigh the cost.